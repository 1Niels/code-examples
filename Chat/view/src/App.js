import {useRef} from 'react';
import { joinChat } from './modules/api';

function App() {
  let username = useRef('');

  async function authenticate(e){
    e.preventDefault();
    console.log(username.current);
    // get the web token from the server
    const token = await joinChat(username.current);
    //store token in session storage
    sessionStorage.setItem('token', token);
  }

  return (
    <div>
      <label>Enter your username</label>
      <input type="text" onChange={(e) => username.current = e.target.value}/>
      <button type="button" onClick={(e) => authenticate(e)}>Join the Chat</button>
        
    </div>
  );
}

export default App;
