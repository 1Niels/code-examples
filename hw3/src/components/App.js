import { React, useState, useRef } from 'react';
import Questions from './Questions';
// import '../assets/App.css';
import questionsList from '../assets/questions.json'

function App() {
  // initiatize state, the game hasnt started yet
  const [start, setStarted] = useState(false);
  //keep track of the state of the game (started or not)
  let username = useRef('');

  function handleStart(){
    setStarted(true); //update the state
    //update start
 
  }

function handleOnChange(e){
    username.current = e.target.value;
}

  if(!start){
    return (
      <div>
        <input type="text" onChange={handleOnChange}/>
        <button type="submit" onClick={handleStart}>Start</button>
        </div>
      );
  }
return (
<div>
  <Questions  questionslist={questionsList} player={username.current}/>
  </div>
);

}

export default App;
