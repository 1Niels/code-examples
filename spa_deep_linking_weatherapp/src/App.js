import React from 'react';
import { BrowserRouter as Router, Link, Route } from 'react-router-dom';
import LoginComponent from './components/LoginComponent';
import WeatherComponent from './components/WeatherComponent';

function WeatherApp() {
  const cities = ['Philadelphia', 'Boston', 'Mumbai', 'Dakar', 'Shanghai', 'London', 'Montreal'];
 
  return (
    <Router>
      <div>
        <section>
          <h1>Cities</h1>
          <nav>
            <ul>
              {cities.map((loc) => (
                <li key={loc}><Link to={`/weather/${loc}`}>{loc}</Link></li>
              ))}
            </ul>
          </nav>

          <article>
          
          <Route path="/weather/:city" render={() => (<WeatherComponent />)} />
            <Route path="/login" render={() => (<LoginComponent />)} />
          </article>
        </section>
      </div>
    </Router>
  );
}


export default WeatherApp;
