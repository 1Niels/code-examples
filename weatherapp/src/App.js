import axios from "axios";

function App() {
  const cities = ['Philadelphia', 'Boston', 'Mumbai', 'Shanghai', 'Dakar'];

  // async/await
  async function getWeather(city){
    try{
      const data = await axios.get(`http://api.openweathermap.org/data/2.5/weather?q=${city}&appid=4d01dbe80c384ea20a6937f2aa98848d`);
      return data

    }catch(err) {
      console.log(err);
    }


  }
  
  function handleClick(e){
    e.preventDefault();
    getWeather(e.target.value).then((resp) => console.log(resp)).catch(err => console.error(err));

  }

  return (
    <div>
     <header>
       <input type="text" />
       <button type="button">OK</button>
       </header>
       <section>
        <h1>Cities</h1>
        <nav>
          <ul>
            {cities.map((city)=>
            <li key={city}><button onClick={handleClick} value={city}>{city}</button></li>)}
          </ul>
        </nav>

       </section>



    </div>
  );
}

export default App;
