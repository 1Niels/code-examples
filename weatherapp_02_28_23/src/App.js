/* eslint-disable react/no-unstable-nested-components */
import React from 'react';
import {
  BrowserRouter as Router, Link, Route, Routes,
} from 'react-router-dom';
import WeatherComponent from './components/WeatherComponent';
import LoginComponent from './components/LoginComponent';
import TypedCityComponent from './components/TypedCityComponent';

function WeatherApp() {
  const cities = ['Philadelphia', 'Boston', 'Mumbai', 'Dakar', 'Shanghai', 'London', 'Montreal', 'Paris'];

  return (
    <Router>
      <div>
        <section>
          <h1>Cities</h1>
          <nav>
            <ul>
              {cities.map((loc) => (
                <li key={loc}><Link to={`/weather/${loc}`}>{loc}</Link></li>
              ))}
              <li key="typed"><Link to="/custom/">Enter a city</Link></li>
            </ul>
          </nav>

          <article>
            <Routes>
              <Route path="/weather/:city" element={<WeatherComponent />} />
              <Route path="/" element={<LoginComponent />} />
              <Route path="/custom/" element={<TypedCityComponent />} />
            </Routes>
          </article>
        </section>
      </div>
    </Router>
  );
}

export default WeatherApp;
