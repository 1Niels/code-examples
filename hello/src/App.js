import './App.css';
import { useState, useRef } from 'react';

function App() {
  // add a state to the component
  const [login, setLogin] = useState(false);
  let name = useRef(''); // reference, its value persists  between rendering
  // button click event handler. 
  const handleClick = (e) =>{
    // update the login state
    setLogin(!login);
    console.log('value', e.target.value);
  }
  // input change event handler
  const handleChange = (e) =>{
    name.current = e.target.value;
    console.log('value', e.target.value);
  }
  // conditional rendering
  if(login === false){
  return (
    <div className="App">
      <label> Name: 
        <input type="text" name='name' onChange={handleChange}/>
      </label>
      <button type='button' onClick={handleClick}>Login</button>
    </div>
  );
  }
  else{
    return (
      <div className="App">
        <label> Welcome  
          {name.current}
        </label>
        <button type='button' onClick={handleClick}>Logout</button>
      </div>
    );

  }
}

export default App;
