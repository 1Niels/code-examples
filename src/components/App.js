import { React, useState, useRef } from 'react';
import Questions from './Questions';
import '../assets/App.css';
import questionsList from '../assets/questions.json'

function App() {
  // initiatize state, the game hasnt started yet
  const [, setStarted] = useState(false);
  //keep track of the state of the game (started or not)
  const start = useRef(false);

  function handleStart(){
    setStarted(true); //update the state
    //update start
    start.current = true;
  }
  if(!start.current){
    return (
      <div>
        <button type="submit" onClick={handleStart}>Start</button>
        </div>
      );
  }
return (
<div>
  <Questions  questionslist={questionsList}/>
  </div>
);

}

export default App;
