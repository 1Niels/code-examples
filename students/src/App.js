import './App.css';
import React, { useState, useRef } from 'react';
import FilterableStudentsTable from './FilterableStudentsTable';
import AddStudent from './AddStudent';

function App() {
  // add a state to the component
  const [login, setLogin] = useState(false);
  const name = useRef(''); // reference, its value persists  between rendering
  
  // lifted state: the list of student
  // we will pass the mutator function as prop
  // to the AddStudent component
  const [students, setStudents] = useState([]);


  // button click event handler.
  const handleClick = (e) => {
    // update the login state
    setLogin(!login);
    console.log('value', e.target.value);
  };
  // input change event handler
  const handleChange = (e) => {
    name.current = e.target.value; // update the reference
    console.log('value', e.target.value);
  };
  // conditional rendering
  if (login === false) {
    return (
      <div className="App">
        <label>
          {' '}
          Name:
          <input type="text" name="name" onChange={handleChange} />
        </label>
        <button type="button" onClick={handleClick}>Login</button>
      </div>
    );
  }

  return (
    <div className="App">
      <label>
        {' '}
        Welcome
        {name.current}
      </label>
      <AddStudent students={students} addNewStudent={setStudents}/>
      <FilterableStudentsTable students={students}/>
      <button type="button" onClick={handleClick}>Logout</button>
    </div>
  );
}

export default App;
