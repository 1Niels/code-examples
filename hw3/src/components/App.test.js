/**
 * @jest-environment jsdom
 */

import  { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import App from './App'

test('start button', () => {
render(<App />);
const linkElement = screen.getByText('Start');
expect(linkElement).toBeInTheDocument();
});


test('snapshot test', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    });
