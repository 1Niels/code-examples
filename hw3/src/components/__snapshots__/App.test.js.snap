// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Start
  </button>
</div>
`;
